weave/workers/helpers.py

```diff
@@ -0,0 +1,203 @@
+import logging
+import os
+from typing import Optional
+
+import aiohttp
+from cachetools import TLRUCache, cached
+from gql import Client, gql
+from gql.transport.requests import RequestsHTTPTransport
+
+from weave.trace_server.trace_server_interface import (
+    CompletionsCreateReq,
+    CompletionsCreateRequestInputs,
+    CompletionsCreateRes,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_internal_service_token() -> str:
+    try:
+        internal_service_token = os.environ["WANDB_INTERNAL_SERVICE_TOKEN"]
+        secret_name = os.environ["WANDB_INTERNAL_SERVICE_TOKEN_SECRET_NAME"]
+    except KeyError:
+        raise KeyError("WANDB_INTERNAL_SERVICE_TOKEN is not set")
+
+    return f"{secret_name}:{internal_service_token}"
+
+
+INTERNAL_SERVICE_TOKEN_PREFIX = "X-Wandb-Internal-Service"
+
+
+def get_authenticated_client(impersonate_as: Optional[str] = None) -> Client:
```

Comment:
for security reasons, I would define a context var that is a Client type that is expected to be set externally. Then the worker setup in our private repository does the work to extract and initialize the authenticated client. Seems a bit risky to put internal authentication details in OSS

Reply:
I would at least do that. IMO, this is really not great to have floating around OSS. Nothing in the OSS repo deals with auth / gorilla interactions. Even as is (now in a public commit) it is outed, merging this to master pretty much locks it in for good in history. 

In this PR: https://github.com/wandb/core/pull/29039/files you are setting calling the `wevae_scorer` entry point directly. I don't see why `main` can't be parameterized to support this injection and in core you have a small script that does the injection.

If you feel this is an ok sec risk, I would like a second opinion.